superset/utils/oauth2.py

```diff
@@ -0,0 +1,69 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+from __future__ import annotations
+
+from datetime import datetime, timedelta
+
+from superset import db
+from superset.db_engine_specs.base import BaseEngineSpec
+
+
+def get_oauth2_access_token(
```

Comment:
Optional: I think in theory a lot of the methods around this could be defined as `async`, which I don't think we're using much around the codebase. Eventually if/when we work on performance improvements for the backend we could look into whether having more async functions defined could help improve perf. Might be great to be able to keep doing other work while we oauth handshake async.

Reply:
for reference: https://flask.palletsprojects.com/en/3.0.x/async-await/#performance

Reply:
Interesting, my comment is out of scope here, resolving